watcher.go

```diff
@@ -234,6 +258,15 @@ func (w *Watcher) Changed(tmplID string) bool {
 	if !initialized { // first pass, always return true
 		return true
 	}
+
+	// Check if the template was buffering and is now finished. We'll return
+	// true since the template had changed at an earlier point but was not ready
+	// then to render then. The cache is not checked because it may have been
+	// cleared by now.
+	if w.bufferTemplates.Buffered(tmplID) {
```

Comment:
Hopefully once the async work is done we'll be able to remove this as how I'm thinking of doing it would eliminate the clearing of the changed/cached info that is used for changed checks.